File: db4oj/core/src/com/db4o/MCommitOK.java
/* Copyright (C) 2004 - 2005  db4objects Inc.  http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;

final class MCommitOK extends Msg {
	public MCommitOK() {
		super();
	}

	public MCommitOK(MsgCloneMarker marker) {
		super(marker);
	}

    final boolean processMessageAtServer(YapSocket in) {
        getTransaction().commit();
        Msg.OK.write(getStream(), in);
        return true;
    }

	public Object shallowClone() {
		return shallowCloneInternal(new MCommitOK(MsgCloneMarker.INSTANCE));
	}
}

File: db4on/core/src/core/com/db4o/cs/messages/MCommitOK.cs
namespace com.db4o.cs.messages
{
	public sealed class MCommitOK : com.db4o.cs.messages.Msg
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 @in)
		{
			GetTransaction().Commit();
			com.db4o.cs.messages.Msg.OK.Write(GetStream(), @in);
			return true;
		}
	}
}
